pheps/phep-0001.md

```diff
@@ -0,0 +1,287 @@
+PHEP: 1
+Title: PHEP Purpose and Guidelines
+Author: Jonathan Niehof <jtniehof@gmail.com>
+Discussions-To: https://github.com/heliophysicsPy/standards/pull/22
+Status: Draft
+Type: Process
+Content-Type: text/markdown; charset=UTF-8; variant=CommonMark
+Created: 20-Oct-2023
+Post-History: 24-Oct-2023
```

Comment:
The preamble looks terrible when rendered. We could make this an indented code block, but RFC 2822 does not (as far as I can tell) allow blank space at the head of the line. We could also set it off in code fences, which again means that we're not strictly starting with the preamble.

Reply:
Maybe this could be a table?

Reply:
Tables are a GH extension, not CommonMark, and would also deviate from the PEP 1 specification of this as RFC 2822. I'm not married to RFC 2822 but minimal deviation (code fence, block indent) seems wise.

Reply:
I agree we should at least get it to render every line on its own line. One simple way to achieve that is to just double space it. Code fence or block indent is a fine idea too. Point is I see no reason to _strictly_ adhere to RFC 2822 (but yes minimal deviations would be nice).

Reply:
Although dang, waffling just a bit, the tables in e.g. PLEPs and SEPs do look really nice...

Reply:
I am persuadable on GH extensions if it seems really compelling, but I'm concerned about tying ourselves too strongly to the GH ecosystem. If there are extensions used elsewhere that GH supports, that's fine.

To the preamble specifically, either code-fencing or indenting sounds good to me. Code-fencing is probably the smallest change and I'll do that in the next push. (Older versions of ReText wordwrap inside codefences but that's definitely a ReText bug). [email.parser](https://docs.python.org/3/library/email.parser.html#module-email.parser) _will_ parse the headers on PHEP-1 but will _not_ parse with either codefence or leading spaces, so I think the codefences are easiest to strip.

Reply:
Fun fact: codefencing the preamble does weird things to the magic number:
```
$ file phep-0001.md
phep-0001.md: Dyalog APL transfer
```

Reply:
Good to know there _is_ a problem with just putting blank lines. The code fence looks nice too. Can you clarify what it does to the magic number? Not sure what you were showing us.

Reply:
This is what it should look like (without the code fence):
```
file phep-0001.md
phep-0001.md: ASCII text, with very long lines (837)
```
The code fence makes `file` [misidentify the file type](https://en.wikipedia.org/wiki/Magic_number_(programming)#In_files). Is this fatal? Not really, just weird...it's called "magic" for a reason.

Reply:
I'm inclined to not worry about the magic number. Modern tech can render files correctly regardless of the magic number, and it looks way better with the code fence, so I vote we use it.

Reply:
I'm declaring this resolved with the code fence.